### PR TITLE
Add template parameters to createNodesOp in scheduler_perf

### PR DESCRIPTION
**Pull request by macsko**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add index and count template parameters to `createNodesOp`. This implementation is similar to `createPodsOp`. Added parameters will be mostly beneficial when using `updateAnyOp` to update previously created nodes.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/framework/perf_utils.go [hunk 1]
```diff
@@ -67,33 +81,18 @@ func (p *IntegrationTestNodePreparer) PrepareNodes(ctx context.Context, nextNode
...
 		var err error
+		var baseNode *v1.Node
 		for retry := 0; retry < retries; retry++ {
 			// Create nodes with the usual kubernetes.io/hostname label.
 			// For that we need to know the name in advance, if we want to
 			// do it in one request.
+			baseNode, err = p.nodeTemplate.GetNodeTemplate(i, numNodes)
+			if err != nil {
+				continue
+			}
```
Comment (sanposhiho) [inline test/integration/framework/perf_utils.go:95]: can be moved to L87?

File: test/integration/framework/perf_utils.go [hunk 2]
```diff
@@ -33,29 +32,44 @@ const (
...
+	// for example a named reference to some other object.
+	GetNodeTemplate(index, count int) (*v1.Node, error)
+}
+
+// StaticNodeTemplate returns an implementation of NodeTemplate for a fixed node that is the same regardless of the index.
+func StaticNodeTemplate(node *v1.Node) NodeTemplate {
+	return (*staticNodeTemplate)(node)
+}
+
+type staticNodeTemplate v1.Node
```
Comment (sanposhiho) [inline test/integration/framework/perf_utils.go:52]: any reason not making it pointer?

```suggestion
type staticNodeTemplate *v1.Node
```

**Review by sanposhiho — COMMENTED** [review]
only nits

File: test/integration/framework/perf_utils.go (hunk 1 above)
Comment (macsko) [reply test/integration/framework/perf_utils.go]: Good idea, moved

**Review by sanposhiho — COMMENTED** [review]
/lgtm

File: test/integration/framework/perf_utils.go [hunk 3]
```diff
@@ -33,29 +32,44 @@ const (
...
+// NodeTemplate is responsible for creating a v1.Node instance that is ready
+// to be sent to the API server.
+type NodeTemplate interface {
+	// GetNodeTemplate returns a node template for one out of many different nodes.
+	// Nodes with numbers in the range [index, index+count-1] will be created
+	// based on what GetNodeTemplate returns. It gets called multiple times
+	// with a fixed index and increasing count parameters. This number can,
+	// but doesn't have to be, used to modify parts of the node spec like
+	// for example a named reference to some other object.
+	GetNodeTemplate(index, count int) (*v1.Node, error)
```
Comment (alculquicondor) [inline test/integration/framework/perf_utils.go:44]: It looks a little odd that you pass two numbers.

What is the use case for the `count` parameter?

File: test/integration/framework/perf_utils.go [hunk 4]
```diff
@@ -33,29 +32,44 @@ const (
 	retries = 5
 )
 
+// NodeTemplate is responsible for creating a v1.Node instance that is ready
+// to be sent to the API server.
+type NodeTemplate interface {
+	// GetNodeTemplate returns a node template for one out of many different nodes.
+	// Nodes with numbers in the range [index, index+count-1] will be created
+	// based on what GetNodeTemplate returns. It gets called multiple times
+	// with a fixed index and increasing count parameters. This number can,
```
Comment (alculquicondor) [inline test/integration/framework/perf_utils.go:41]: it seems to be the opposite?

File: test/integration/framework/perf_utils.go [hunk 5]
```diff
@@ -33,29 +32,44 @@ const (
 	retries = 5
 )
 
+// NodeTemplate is responsible for creating a v1.Node instance that is ready
+// to be sent to the API server.
+type NodeTemplate interface {
+	// GetNodeTemplate returns a node template for one out of many different nodes.
+	// Nodes with numbers in the range [index, index+count-1] will be created
```
Comment (alculquicondor) [inline test/integration/framework/perf_utils.go:39]: This is a bit confusing. It's not from this range, it's just the number `index`, but the node belongs to a group of `count` nodes.

File: test/integration/framework/perf_utils.go
Comment (alculquicondor) [inline test/integration/framework/perf_utils.go:1]: Before we add more changes here, can we move the code to scheduler_perf?

File: test/integration/framework/perf_utils.go (hunk 3 above)
Comment (macsko) [reply test/integration/framework/perf_utils.go:44]: These two are typical template parameters for other operations (createPods, createAny, updateAny). For pods it is useful for scaling the resource requests to fill up the nodes ([ref](https://github.com/kubernetes/kubernetes/blob/9ffc095f88c74c0f5b69b1c4a645836f0b7c83fc/test/integration/scheduler_perf/config/event_handling/podupdate-pod-noderesources.yaml#L14)). For nodes, we could think of something similar I think.

File: test/integration/framework/perf_utils.go (hunk 5 above)
Comment (macsko) [reply test/integration/framework/perf_utils.go]: Right, changed

File: test/integration/framework/perf_utils.go
Comment (macsko) [reply test/integration/framework/perf_utils.go:1]: Done

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**